### PR TITLE
Add primeng@2.0.1 w/ npm auto-update

### PR DESCRIPTION
**Pull request by hare1039**
Pull request for issue: #10134
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/primefaces/primeng
 * Official website (optional, not the repository): http://www.primefaces.org/primeng/#/
 * NPM package url (optional): https://www.npmjs.com/package/primeng
 * License and its reference: MIT, https://github.com/primefaces/primeng/blob/master/LICENSE.md
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 177
   - Count of stars: 1622
   - Count of forks: 555
 * NPM download stats (optional):
   - Downloads in the last day: 2862
   - Downloads in the last week: 17719
   - Downloads in the last month: 64018

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
Any reference of this library that cdnjs should host?
I saw the `primeng.js` include `components` folder.

**Comment by hare1039** [comment]
Sorry I forgot to mention  https://github.com/primefaces/primeng/issues/2118

File: ajax/libs/primeng/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "homepage": "http://www.primefaces.org/primeng/",
+  "author": "cagataycivici",
+  "license": "MIT",
+  "npmName": "primeng",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "primeng.js*",
+        "resources/**/!(*.scss)"
```
Comment (extend1994) [inline ajax/libs/primeng/package.json]: As the issue about what should we host you mentioned, I think we only need two themes (bootstrap and omega). Here you include so many other themes' files. Also, they are added in this commit, please remove them.

Comment (hare1039) [inline ajax/libs/primeng/package.json]: @extend1994 I discuss with @pvnr0082t about this, and I think we should not only fetch only two themes' `css`.

@pvnr0082t Would you give some advice about this situation? 
And please help me review on this PR. Thanks.

Comment (pvnr0082t) [inline ajax/libs/primeng/package.json]: @hare1039 Please confirm this with author, Thanks.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by hare1039** [comment]
Nope response from the author.
I think we just add all theme files.

**Comment by hare1039** [comment]
@pvnr0082t I discover that this PR should be a high priority issue. Can we add this library first?

File: ajax/libs/primeng/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "datagrid",
+    "charts",
+    "typescript"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/primefaces/primeng.git"
+  },
+  "homepage": "http://www.primefaces.org/primeng/",
+  "author": "cagataycivici",
```
Comment (pvnr0082t) [inline ajax/libs/primeng/package.json]: I think it's better to use `Çağatay Çivici`. How do you think?

Comment (PeterDaveHello) [inline ajax/libs/primeng/package.json]: @hare1039 it this added by your own `"author": "cagataycivici",`?

Comment (hare1039) [inline ajax/libs/primeng/package.json]: @pvnr0082t I fix `Çağatay Çivici`. Please review this PR again. Thanks!

Comment (PeterDaveHello) [inline ajax/libs/primeng/package.json]: @hare1039 it this added by your own "author": "cagataycivici",?

**Comment by pvnr0082t** [comment]
@hare1039 Please rebase with the latest master branch. Thanks.

**Comment by hare1039** [comment]
ping

**Comment by PeterDaveHello** [comment]
@hare1039 be polite.

**Comment by PeterDaveHello** [comment]
@hare1039 I have no idea why would you leave my question alone but ping, what's the situation here?

https://github.com/cdnjs/cdnjs/pull/10749#discussion_r111449563

**Comment by PeterDaveHello** [comment]
Please also confirm that we do need `primeng.js`, take a look at its content, the resource been required, are not added, and take a look at the doc:

https://www.primefaces.org/primeng/#/setup

Maybe its js file is not used this way.

**Comment by sufuf3** [comment]
@hare1039 Would you reopen primefaces/primeng#2118
And try to discuss with the upstream again?
Thank you.

**Comment by sashberd** [comment]
@MattIPv4 I need you suggestion about globs for this one

**Comment by sashberd** [comment]
@MattIPv4 I think we cannot add this lib to cdnjs cause most files in new versions are *.ts (typescript) extentions.
Only resources folder has *.css files.
FYR

**Comment by MattIPv4** [comment]
Typescript is supported, so this will be fine to add once the PR is fixed to only be a package.json file.

**Comment by sashberd** [comment]
@MattIPv4 So in this case we will need to add all files and folders from base, except *.json and readme.
Am I right?

**Comment by MattIPv4** [comment]
I'm not sure we actually need the TS files.

I think we just need the JS files from:
`bundles`, `esm2015`, `fesm2015`, `fesm5`.

As well as all the files in `resources`?

**Comment by sashberd** [comment]
Moved to #13723

**Pull request closed without merging**